### PR TITLE
Implement sub_42D8FF

### PR DESCRIPTION
**Pull request by GalBr**
This is a sub function for the removeBuilding game command.
I'll probably implement the game command itself as well and then remove the hook I added here.
I tested this code by removing buildings in towns with the clear area tool.

### CODE REVIEW

File: src/OpenLoco/Map/TileManager.cpp [hunk 1]
```diff
@@ -713,6 +715,63 @@ namespace OpenLoco::Map::TileManager
...
+        regs.cx = pos.x;
+        regs.dx = pos.y;
+        regs.bp = pos.z;
+        call(0x0048B0C7, regs);
+    }
+
+    // 0x0042D8FF
+    void sub_42D8FF(TileElement* element, const Map::Pos2& pos)
+    {
+        auto* buildingEl = element->as<BuildingElement>();
```
Comment (duncanspumpkin) [inline src/OpenLoco/Map/TileManager.cpp:731]: in general we have been calling the variable `elBuilding`

File: src/OpenLoco/Map/TileManager.cpp [hunk 2]
```diff
@@ -713,6 +715,63 @@ namespace OpenLoco::Map::TileManager
...
+        if (buildingEl == nullptr)
+            return;
+
+        if (!buildingEl->isGhost() && !buildingEl->isFlag5())
+        {
+            if (CompanyManager::getUpdatingCompanyId() != CompanyId::neutral)
+            {
+                sub_48B0C7(Map::Pos3(pos.x + 16, pos.y + 16, buildingEl->baseZ() * 4));
```
Comment (duncanspumpkin) [inline src/OpenLoco/Map/TileManager.cpp:739]: `createDestructExplosion`

File: src/OpenLoco/Map/TileManager.cpp [hunk 3]
```diff
@@ -713,6 +715,63 @@ namespace OpenLoco::Map::TileManager
...
+        if (!buildingEl->multiTileIndex())
+        {
+            if (!buildingEl->isGhost())
+            {
+                auto* buildingObj = buildingEl->getObject();
+                if (buildingObj != nullptr)
+                {
+                    if (!(buildingObj->flags & BuildingObjectFlags::misc_building))
+                    {
+                        auto population = -buildingObj->producedQuantity[0];
```
Comment (duncanspumpkin) [inline src/OpenLoco/Map/TileManager.cpp:752]: `buildingCapacity`

File: src/OpenLoco/Map/TileManager.cpp [hunk 4]
```diff
@@ -713,6 +715,63 @@ namespace OpenLoco::Map::TileManager
...
+    {
+        registers regs;
+        regs.cx = pos.x;
+        regs.dx = pos.y;
+        regs.bp = pos.z;
+        call(0x0048B0C7, regs);
+    }
+
+    // 0x0042D8FF
+    void sub_42D8FF(TileElement* element, const Map::Pos2& pos)
```
[the remark this answers is not in the document]
Comment (duncanspumpkin) [reply src/OpenLoco/Map/TileManager.cpp:729]: What about removeBuildingElement for the name

[the remark this answers is not in the document]
Comment (GalBr) [reply src/OpenLoco/Map/TileManager.cpp]: I actually started with `BuildingElement&` and changed it when I tried to find why the code crashed. Forgot to switch back to it. Thanks!

File: src/OpenLoco/Map/TileManager.cpp [hunk 5]
```diff
@@ -713,6 +715,63 @@ namespace OpenLoco::Map::TileManager
...
+
+        if (!buildingEl->isGhost() && !buildingEl->isFlag5())
+        {
+            if (CompanyManager::getUpdatingCompanyId() != CompanyId::neutral)
+            {
+                sub_48B0C7(Map::Pos3(pos.x + 16, pos.y + 16, buildingEl->baseZ() * 4));
+            }
+        }
+
+        if (!buildingEl->multiTileIndex())
```
[the remark this answers is not in the document]
Comment (GalBr) [reply src/OpenLoco/Map/TileManager.cpp]: I forget this is not a a bool function every time, thanks!

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**